src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1583,9 +1583,12 @@ static int client_parse_message(
 
         switch (client->state) {
         case DHCP_STATE_SELECTING:
-                if (r != DHCP_OFFER)
+                if (r == DHCP_ACK && !lease->rapid_commit)
```

Comment:
I have not read the RFC yet, but maybe we should also check that we requested rapid commit option in DISCOVER message.

Reply:
Yeah I just read it too after finding the rapid_commit option on openwrt wiki. Thankfully it's pretty short.

I guess we probably should reject a rapid ack if we did not request it, I'll add that.